apis/src/components/atoms/history_button.rs

```diff
@@ -73,4 +74,48 @@ fn send_action(action: &HistoryNavigation) {
             game_state_signal.view_game()
         }
     }
+    set_timer_from_response();
+}
+
+pub fn set_timer_from_response() {
+    let game_state_signal = expect_context::<GameStateSignal>();
+    let timer = expect_context::<TimerSignal>();
+    let is_finished = game_state_signal.is_finished();
+    //don't update timer if game isnt finished or for correspondence games
+    if !is_finished()
+        || matches!(
+            game_state_signal.signal.get().get_game_speed(),
+            None | Some(GameSpeed::Correspondence) | Some(GameSpeed::Untimed)
+        )
+    {
+        return;
+    }
+    let turn = game_state_signal
```

Comment:
It's probably worth getting the signal just once and then using it in all the places it's currently got.